goethe/corpora.py

```diff
@@ -1,168 +1,44 @@
+        return self.tokens if self.use_tokens else self.sents
 
-    def delete_word_index(self):
-        """Delete word indices built with `build_word_index()`.
+    @property
+    def sents(self):
+        """Yield sentences from file.
         """
-        self.word2index = None
-        self.index2word = None
-
-    def preprocess_and_store(self, directory, log=None):
-        if not os.path.exists(directory):
-            os.makedirs(directory)
-
-        self.store_tokens(os.path.join(directory, 'tokens.txt'), log=log)
-        self.store_numbers(os.path.join(directory, 'numbers.txt'), log=log)
-        self.store_index2word(os.path.join(directory, 'index2word.txt'))
-
-    def store_tokens(self, filename, log=None):
-        with open(filename, 'w') as f:
-            for i, sentence in enumerate(self.sentences(words=True)):
-                if log and i % log == 0:
-                    print("tokens - wrote %d lines" % i)
-                f.write(" ".join(sentence) + '\n')
-
-    def store_numbers(self, filename, log=None):
-        with open(filename, 'w') as f:
-            for i, numbers in enumerate(self.number_sentences()):
-                if log and i % log == 0:
-                    print("numbers - wrote %d lines" % i)
-                f.write(" ".join(map(str, numbers)) + '\n')
+        path = self.file_path(use_tokens=False)
+        with open(path) as f:
+            yield from (l.strip() for l in f)
 
-    def store_triples(self, filename, log=None):
-        with open(filename, 'w') as f:
-            for i, ngram in self.number_ngrams:
-                if log and i % log == 0:
-                    print("triples - wrote %d lines" % i)
-                f.write(" ".join(map(str, ngram)) + "\n")
-
-    def store_index2word(self, filename):
-        if not self.word2index:
-            self.build_word_index()
-        with open(filename, 'w') as f:
-            for word in self.index2word:
-                f.write(word + '\n')
-
-    def number_sentences(self):
-        if not self.word2index:
-            self.build_word_index()
-        for sentence in self.sentences(words=True):
-            yield [self.word2index[word] for word in sentence]
-
-    def rolling_window(self, a, window):
-        shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
-        strides = a.strides + (a.strides[-1],)
-        return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
-
-    def number_ngrams(self, n):
-        for numbers in self.number_sentences():
-            for ngram in self.rolling_window(np.array(numbers), n):
-                yield ngram
-
-    def number_ngrams_(self, n):
-        for numbers in self.number_sentences():
-            for i in range(0, len(numbers) - n + 1):
-                yield numbers[i:i+n]
-
-    def sentences(self, words=True):
-        """Iterate over lines in files returning sentences.
+    @property
+    def tokens(self):
```

Comment:
Correct me if I am wrong, but for me it seems not right to mark an generator as a property (I don't know the pythonic way).